Tasks/Common/codecoverage-tools/cobertura/cobertura.ant.ccenabler.ts

```diff
@@ -0,0 +1,195 @@
+/// <reference path="../../../../definitions/Q.d.ts" />
+/// <reference path="../../../../definitions/string.d.ts" />
+/// <reference path="../../../../definitions/vsts-task-lib.d.ts" />
+/// <reference path="../../../../definitions/node.d.ts" />
+
+import * as Q from "q";
+import * as util from "../utilities";
+import * as tl from "vsts-task-lib/task";
+import * as ccc from "../codecoverageconstants";
+import * as cc from "../codecoverageenabler";
+import * as str from "string";
+import * as os from "os";
+import * as path from "path";
+
+export class CoberturaAntCodeCoverageEnabler extends cc.CoberturaCodeCoverageEnabler {
+
+    reportDir: string;
+    reportbuildfile: string;
+    classDirs: string;
+    includeFilter: string;
+    excludeFilter: string;
+    sourceDirs: string;
+
+    // -----------------------------------------------------
+    // Enable code coverage for Cobertura Ant Builds
+    // - enableCodeCoverage: CodeCoverageProperties  - ccProps
+    // -----------------------------------------------------    
+    public enableCodeCoverage(ccProps: { [name: string]: string }): Q.Promise<boolean> {
+        let _this = this;
+
+        tl.debug("Input parameters: " + JSON.stringify(ccProps));
+
+        _this.reportDir = "CCReport43F6D5EF";
+        _this.reportbuildfile = "CCReportBuildA4D283EG.xml";
+        _this.buildFile = ccProps["buildfile"];
+        let classFilter = ccProps["classfilter"];
+        let srcDirs = ccProps["sourcedirectories"];
+        if (str(srcDirs).isEmpty()) {
+            srcDirs = ".";
+        }
+        _this.sourceDirs = srcDirs;
+        _this.classDirs = ccProps["classfilesdirectories"];
+
+        let filter = _this.extractFilters(classFilter);
+        _this.excludeFilter = _this.applyFilterPattern(filter.excludeFilter).join(",");
+        _this.includeFilter = _this.applyFilterPattern(filter.includeFilter).join(",");
+
+        tl.debug("Reading the build file: " + _this.buildFile);
+
+        return util.readXmlFileAsJson(_this.buildFile).
+            then(function (resp) {
+                return _this.addCodeCoverageData(resp);
+            })
+            .thenResolve(true);
```

Comment:
You're returning a promise but already resolving it.  That's breaking the chain for the consumer.  Why thenResolve(true)?